feat(text-field): add types and props

max-umain opened the pull request:
## **Describe pull-request**  
Adds "email" and "tel" as possible values of the type prop, as well as the props required and autocomplete.

## **Issue Linking:**  
- **Jira:** [CDEP-1526](https://jira.scania.com/browse/CDEP-1526)

## **How to test** 
1. Go to https://pr-1465.d3fazya28914g3.amplifyapp.com/?path=/story/components-text-field--default
2. Try setting the type prop to "email" and "tel", and setting the required and autocomplete props, and verify by inspecting with the browser dev tools that these are applied as attributes to the `<input>` element

## **Checklist before submission**
- [ ] Designer approves new design (if applicable)
- [ ] No accessibility violations in Storybook
- [ ] I have added unit tests for my changes (if applicable)
- [ ] All existing tests pass
- [ ] I have updated the documentation (if applicable)
- [ ] Not breaking production behavior
- [ ] Behavior available in Storybook with documented descriptions (if applicable)
- [ ] `npm run build:all` without errors

## **Suggested test steps**
- [ ] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [ ] Storybook controls
- [ ] Design/controls/props is aligned with other components 
- [ ] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [ ] Events

## **Screenshots**  
None

## **Additional context**  
None

Lunkan89 reviewed (commented):
Nice, should we test that required works in form, both e2e and demo page

ckrook reviewed (changes requested):
I agree with Johan, I would like to test this on a demo page (react) with an actual mobile device 😊

max-umain commented:
Just for documentation: I added tests for required on all components that have it as an option. However, the screenshot differs due to a different text style being used in the "popup warning" in the one being run in github:

<img width="1280" height="720" alt="form-test-click-submit-with-missing-required-inputs-1-expected" src="https://github.com/user-attachments/assets/c2969f1f-f5d3-4898-b5d0-89001034b855" />

<img width="1280" height="720" alt="form-test-click-submit-with-missing-required-inputs-1-actual" src="https://github.com/user-attachments/assets/1e67622e-85f6-414c-85d1-d37f172170a4" />

So I changed it now to check the missingValue attribute on the HTMLInputElements

max-umain commented on packages/core/src/components/text-field/text-field.tsx:
```diff
@@ -77,6 +77,12 @@ export class TdsTextField {
+  @Prop() required: boolean = false;
+
+  /** Value to be used for the text field's autocomplete attribute */
+  @Prop() autocomplete: string = 'off';
```
I thought about that, but the problem is it might make users think these are the only possible values for the autocomplete attribute, which isn't the case as there are many more than those. What do you think about this?

ckrook replied:
Good point you are right

ckrook reviewed (approved):
Really great that you added tests to this feature 🚀

maxadolfsson commented:
@max-umain @Lunkan89 @ckrook – We had a question about file upload in the design support channel, I would assume that would make sense to do in another ticket/PR and not squeeze it in here? Let me know what you think!
